portal/i18next_gulpfile.js

```diff
@@ -0,0 +1,186 @@
+                        list: ['i18next.t', 'i18n.t'],
+                        extensions: ['.js', '.jsx']
+                    }
+                    ,
+                    resource: {
+                        //the source path is relative to current working directory as specified in the destination folder
+                        savePath: './src/' + nameSpace + '.json'
+                    }
+                }))
+              .pipe(gulp.dest('translations/js'));
+});
+
+/*
+ * convert json to pot (the definition file) for translator's consumption
+ */
+gulp.task('i18nextConvertJSONToPOT', ['i18next-extraction'], function() {
+
+    const options = {/* you options here */}
+   /*
+    * converting json to pot
+    */
+    console.log("converting JSON to POT...");
+    del(['translations/js/src/translation.pot']);
+    i18nextConv.i18nextToPot('en', fs.readFileSync(translationSourceDir+nameSpace+".json"), options).then(save(translationSourceDir+nameSpace+".pot"));
+
+});
+
+/*
+ * combine newly created pot file to existing messages.pot file ???
+ * do we need this step??
+ */
+gulp.task('combineAllPotFiles', ['i18nextConvertJSONToPOT'], function() {
+    console.log("combine all pot files ...")
+    return gulp.src([translationSourceDir+nameSpace+".pot", 'translations/messages.pot'])
+          .pipe(concatPo('messages.pot'))
+          .pipe(gulp.dest('translations'));
+});
+
+/*
+ * converting po to json files
+ * note translating existing po file to json, which will be consumed by the front end
+ * this task assumes that:
+ *    1. text has been extracted from js file into JSON file
+ *    2. translated JSON into POT
+ *    3. merged new POT into main POT file [need to check about this step]
+ *    4. Po files have been returned from translator after uploading POT file from #3
+ */
+gulp.task('i18nextConvertPOToJSON', function() {
+  console.log("converting po to json ...")
+  const options = {/* you options here */}
+
+  del(['static/files/**/translation.json']);
+
+   /*
+    * translating po file to json for supported languages
+    */
+  languagesArray.forEach(function(lng) {
+      var destination = translationDestinationDir+lng+"/translation.json";
+      fs.open(destination, 'w+', function(err, fd) {
+        if (err) console.log("error occurred writing " + lng + " json: " + err);
+        else {
+          i18nextConv.gettextToI18next(lng, fs.readFileSync(path.join(__dirname,'./translations/' + lng.replace('-', '_') + '/LC_MESSAGES/messages.po')), options)
+          .then(save(destination));
+          fs.close(fd, function(err) {
+              if (err) console.log("error occurred closing " + destination);
+          });
+        };
+      });
+  });
+});
+
+/*
+ * concating all necessary i18next JS files into one
+ * don't run this unless files have been changed
+ */
+gulp.task('i18nextConcatScripts', function() {
+  console.log("concat i18next scripts ...")
+  var i18nextMain = request('https://unpkg.com/i18next/i18next.js')
```

Comment:
Can these libraries be used directly from unpkg.com? unpkg appears to be a CDN so they be able to better serve users in more remote locations.

Reply:
yup, will remove this task - it was meant for me to test.  The libraries are loaded directly unpkg/cdn (see changes in layout.html)